File: fm-domain/src/main/java/io/zeitwert/fm/building/service/api/dto/EvaluationBuilding.java
package io.zeitwert.fm.building.service.api.dto;

import java.awt.Color;

import io.zeitwert.fm.util.Formatter;
import lombok.Builder;
import lombok.Data;

@Data()
@Builder
public class EvaluationBuilding implements Comparable<EvaluationBuilding> {

	private Integer id;
	private String name;
	private String description;
	private String buildingNr;
	private String address;

	private Integer insuredValue;
	private Integer relativeValue;
	private Integer insuredValueYear;

	private Integer ratingYear;

	private Integer condition; // zn100
	private Color conditionColor;

	public String getFormattedInsuredValue() {
		return Formatter.INSTANCE.formatNumber(1000 * this.insuredValue);
	}

	public String getRelativeValue() {
		int weight = (int) Math.round(this.relativeValue * 70 / 100);
		return new String(new char[weight]).replace('\0', 'I');
	}

	@Override
	public int compareTo(EvaluationBuilding o) {
		int res = this.buildingNr.compareTo(o.buildingNr);
		if (res == 0) {
			res = this.name.compareTo(o.name);
		}
		return res;
	}

}

File: fm-domain/src/main/java/io/zeitwert/fm/building/api/dto/EvaluationBuilding.kt
package io.zeitwert.fm.building.api.dto

import io.zeitwert.fm.util.Formatter
import java.awt.Color
import kotlin.math.roundToInt

data class EvaluationBuilding(
	val id: Int,
	val name: String?,
	val description: String?,
	val buildingNr: String?,
	val address: String?,
	val insuredValue: Int?,
	val relativeValue: Int,
	val insuredValueYear: Int,
	val ratingYear: Int,
	val condition: Int,
	val conditionColor: Color,
) : Comparable<EvaluationBuilding> {

	val formattedInsuredValue: String
		get() = Formatter.formatNumber(1000 * (insuredValue ?: 0))

	fun getRelativeValue(): String {
		val weight = (relativeValue * 70.0 / 100).roundToInt()
		return String(CharArray(weight)).replace('\u0000', 'I')
	}

	override fun compareTo(other: EvaluationBuilding): Int {
		var res = buildingNr!!.compareTo(other.buildingNr!!)
		if (res == 0) {
			res = name!!.compareTo(other.name!!)
		}
		return res
	}

}
